sonaranalyzer-dotnet/rspec/cs/S1192_c#.html

```diff
@@ -35,5 +35,6 @@ <h2>Exceptions</h2>
 <ul>
   <li> literals with fewer than 5 characters </li>
   <li> literals matching one of the parameter names </li>
+  <li> literals used in attributes </li>
```

Comment:
Should we restrict only to certain attributes? Like Obsolete, SuppressMessage, etc.? Attribute strings could be constants declared in the same class and sometimes it makes sense to avoid having literals...

Reply:
I thought about restricting it to known attributes but the problem was deciding which ones. There are a lot of attributes in the _System.Diagnostics_ namespace that take strings, and strings are commonly used with MEF attributes too. It seemed that it would be difficult to come up with a comprehensive list, so we'd have to revisit this rule repeatedly to add more cases -> reasonable likelihood of other false positives.

Looking at it the other way, there aren't many cases in which you would validly want to define a constant and use it in an attribute. Ids for Visual Studio packages are one, but the id is only used in two places so it wouldn't be caught by the default rule in any case. I couldn't think of any compelling cases in which you'd define a constant and use it in enough places in an attribute and the code to trigger the rule -> low likelihood of false negatives.